### PR TITLE
Replace STCxxx with STC enum

### PR DESCRIPTION
**Pull request by wilzbach**
The second commit was done automatically:

```shell
replacements=(
"STCautoref autoref"
"STCinference inference"
"STCinit init"
"STCscopeinferred scopeinferred"
"STCundefined undefined_"
"STCstatic static_"
"STCextern extern_"
"STCconst const_"
"STCfinal final_"
"STCabstract abstract_"
"STCparameter parameter"
"STCfield field"
"STCoverride override_"
"STCauto auto_"
"STCsynchronized synchronized_"
"STCdeprecated deprecated_"
"STCin in_"
"STCout out_"
"STClazy lazy_"
"STCforeach foreach_"
"STCvariadic variadic"
"STCctorinit ctorinit"
"STCtemplateparameter templateparameter"
"STCscope scope_"
"STCimmutable immutable_"
"STCref ref_"
"STCmanifest manifest"
"STCnodtor nodtor"
"STCnothrow nothrow_"
"STCpure pure_"
"STCtls tls"
"STCalias alias_"
"STCshared shared_"
"STCgshared gshared"
"STCwild wild"
"STCproperty property"
"STCsafe safe"
"STCtrusted trusted"
"STCsystem system"
"STCctfe ctfe"
"STCdisable disable"
"STCresult result"
"STCnodefaultctor nodefaultctor"
"STCtemp temp"
"STCrvalue rvalue"
"STCnogc nogc"
"STCvolatile volatile_"
"STCreturn return_"
"STCexptemp exptemp"
"STCmaybescope maybescope"
"STCfuture future"
"STClocal local"
"STC_TYPECTOR TYPECTOR"
"STC_FUNCATTR FUNCATTR"
)
for r in "${replacements[@]}" ; do
    w=($r)
    sed "s/${w[0]}/STC.${w[1]}/g" -i **/*.d
done
```

### CODE REVIEW

**Comment by wilzbach** [comment]
Recent precedents: https://github.com/dlang/dmd/pull/7319 or https://github.com/dlang/dmd/pull/7328

**Comment by WalterBright** [comment]
"The build failed on Semaphore" I clicked on "Details" and could not find any information on what failed.

**Comment by wilzbach** [comment]
> "The build failed on Semaphore" I clicked on "Details" and could not find any information on what failed.

- I don't exactly know what failed, either but it seems to work know
- I agree that their UI isn't the best. You can expand the individual build run to see the build run:

image

As mentioned in https://github.com/dlang/dmd/pull/7617, this is just an experiment motivated by (1) Travis increasing failures and (2) the fact that SemaphoreCI runs the entire test-suite 2x (with a host compiler and self-bootstrapped)  _and_ Phobos in 8 minutes (Travis takes about 25 mins).

File: src/dmd/astbase.d [hunk 1]
```diff
@@ -107,64 +107,68 @@ struct ASTBase
...
+        inference           = (1L << 46),   // do attribute inference
+        exptemp             = (1L << 47),   // temporary variable that has lifetime restricted to an expression
+        maybescope          = (1L << 48),   // parameter might be 'scope'
+        scopeinferred       = (1L << 49),   // 'scope' has been inferred and should not be part of mangling
+        future              = (1L << 50),   // introducing new base class function
+        local               = (1L << 51),   // do not forward (see dmd.dsymbol.ForwardingScopeDsymbol).
+
+        TYPECTOR = (STC.const_ | STC.immutable_ | STC.shared_ | STC.wild),
+        FUNCATTR = (STC.ref_ | STC.nothrow_ | STC.nogc | STC.pure_ | STC.property | STC.safe | STC.trusted | STC.system),
+    }
```
Comment (wilzbach) [inline src/dmd/astbase.d]: FWIW I'm still not a huge friend of this code duplication in `ast*`

**Pull request merged**